cli/cache/mod.rs

```diff
@@ -146,9 +169,17 @@ impl Loader for FetchCacher {
             Err(err)
           },
           |file| {
+            let maybe_overridden_headers =
+              file_header_overrides.get(&specifier);
+
+            let maybe_headers = maybe_extend_optional_map(
+              file.maybe_headers.as_ref(),
+              maybe_overridden_headers,
+            );
```

Comment:
Perhaps we should make `file_header_overrides` an `Option` and that way we can avoid cloning and do something like this?

```suggestion
            let maybe_headers = match maybe_file_header_overrides.and_then(|o| o.get(&specifier)) {
              Some(overrides) => extend_optional_map(
                file.maybe_headers,
                overrides,
              ),
              None => file.maybe_headers
            };
```

Reply:
I'm not a fan of optional containers, because the meaning of `Some(empty_container)` is often unclear.  
In your suggested change, there'd still be a clone of `file.maybe_headers` if `overrides` was empty. I've gotten rid of `maybe_extend_optional_map()` and replaced it with an inline tuple `match`, that should have avoided the clone.